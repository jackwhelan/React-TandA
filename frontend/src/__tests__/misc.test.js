import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';

import NavBar from '../components/AppNavbar.component';
import News from '../components/News.component';
import Schedule from '../components/Schedule.component';

describe('Miscellaneous Components', () => {
    test('Navigation Bar', () => {
        const component = renderer.create(<Router><NavBar /></Router>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('News', () => {
        const component = renderer.create(<News />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Schedule', () => {
        const component = renderer.create(<Schedule />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})