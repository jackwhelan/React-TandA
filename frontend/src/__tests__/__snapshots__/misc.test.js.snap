// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Miscellaneous Components Navigation Bar 1`] = `
<div>
  <nav
    className="bg-navBlue navbar navbar-expand-sm navbar-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        TIMA 
        <span
          className="text-muted"
        >
          | 
          Home
        </span>
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="mr-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-home"
              />
               Home
            </a>
          </li>
        </ul>
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Login"
              onClick={[Function]}
            >
              <i
                className="fas fa-sign-in-alt"
              />
               Log In
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Register"
              onClick={[Function]}
            >
              <i
                className="fas fa-user-plus"
              />
               Register
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Miscellaneous Components News 1`] = `
<div
  className="core-cont bg-lightgrey mt-4"
>
  <h2>
    Loading...
  </h2>
  <hr
    className="bg-black"
  />
  <p>
    We are attempting to load news items from the database, if this problem persists, please contact a system administrator.
  </p>
</div>
`;

exports[`Miscellaneous Components Schedule 1`] = `
<div
  className="core-cont bg-lightgrey mt-4"
>
  <h2>
    Loading...
  </h2>
  <hr
    className="bg-black"
  />
  <p>
    We are attempting to load events from the database, if this problem persists, please contact a system administrator.
  </p>
</div>
`;
